### PR TITLE
Table polish

### PR DESCRIPTION
**Pull request by willmcgugan**
- Adds a auto sized rows to DataTable
- Adds table to widgets

### CODE REVIEW

File: src/textual/widgets/_data_table.py [hunk 1]
```diff
@@ -55,10 +53,20 @@ class Column:
...
+    width: int = 0
     visible: bool = False
     index: int = 0
 
+    content_width: int = 0
+    auto_width: bool = False
+
+    @property
+    def render_width(self) -> int:
+        if self.auto_width:
```
Comment (darrenburns) [inline src/textual/widgets/_data_table.py:66]: Could you add a comment to the code saying what the `+ 2` is for future readers?

File: src/textual/widgets/_data_table.py [hunk 2]
```diff
@@ -263,34 +276,61 @@ def _get_cell_region(self, row_index: int, column_index: int) -> Region:
...
         height = row.height
         y = row.y
         if self.show_header:
             y += self.header_height
         cell_region = Region(x, y, width, height)
         return cell_region
 
-    def add_column(self, label: TextType, *, width: int = 10) -> None:
+    def add_columns(self, *labels: TextType) -> None:
+        """Add a number of columns:
```
Comment (darrenburns) [inline src/textual/widgets/_data_table.py]:
```suggestion
        """Add a number of columns.
```

File: src/textual/widgets/_data_table.py [hunk 3]
```diff
@@ -299,10 +340,34 @@ def add_row(self, *cells: CellType, height: int = 1) -> None:
...
         self.row_count += 1
         self._line_no += height
-        self._update_dimensions()
-        self.refresh()
+
+        self._new_rows.add(row_index)
+        self._require_update_dimensions = True
+        self.check_idle()
+
+    def add_rows(self, rows: Iterable[Iterable[CellType]]) -> None:
```
Comment (darrenburns) [inline src/textual/widgets/_data_table.py:348]: Why the difference in API with this vs `add_columns` (`Iterable` vs `*`)?

**Review by darrenburns — APPROVED** [review]

File: src/textual/widgets/_data_table.py (hunk 2 above)
Comment (willmcgugan) [reply src/textual/widgets/_data_table.py]: Spotted that one.

**Pull request merged**